client/hotrod-client/src/main/java/org/infinispan/client/hotrod/MetadataValue.java

```diff
@@ -7,32 +7,6 @@
  * @author Tristan Tarrant
  * @since 5.2
  */
-public interface MetadataValue<V> extends VersionedValue<V> {
-
-   /**
-    *
-    * @return Time when entry was created. -1 for immortal entries.
-    */
-   long getCreated();
-
-   /**
-    *
-    * @return Lifespan of the entry in seconds. Negative values are interpreted as unlimited
-    *         lifespan.
-    */
-   int getLifespan();
-
-   /**
-    *
-    * @return Time when entry was last used. -1 for immortal entries.
-    */
-   long getLastUsed();
-
-   /**
-    *
-    * @return The maximum amount of time (in seconds) this key is allowed to be idle for before it
-    *         is considered as expired.
-    */
-   int getMaxIdle();
+public interface MetadataValue<V> extends VersionedValue<V>, Metadata {
```

Comment:
What's the exact reason to separate metadata stuff out? The reason I ask is because although compile-time compatibility is maintained, I'm not sure binary compatibility is maintained with this change.